src/components/AddressSearch/index.tsx

```diff
@@ -270,8 +284,39 @@ function AddressSearch(
                 timeout: 30000,
             },
         );
-    };
+    }, [isFetchingCurrentLocation, onPress]);
+    const selectFocusedOption = useCallback(() => {
+        if (focusedIndex === 0 && shouldShowCurrentLocationButton) {
+            getCurrentLocation();
+            return;
+        }
+        if (!resultRef.current) {
+            return;
+        }
 
+        const data = resultRef?.current[shouldShowCurrentLocationButton ? focusedIndex - 1 : focusedIndex] as GooglePlaceData;
+        if (!data) {
+            return;
+        }
+        // eslint-disable-next-line prefer-template
+        const url = '/api/Proxy_GooglePlaces?proxyUrl='.concat(`/place/details/json?place_id=${data?.place_id}`);
```

Comment:
@jsdev2547 Why are you making a API call here? To fix the personal details search results breakage
The requirement is to only add a keyboard navigation to existing results list functionality.

If you have this https://github.com/Expensify/App/pull/41035#issuecomment-2091547844 analysis, kindly provide it in an understandable format as mentioned here https://github.com/Expensify/App/pull/41035#issuecomment-2084540729

Kindly discuss with C+ or internal engineer before making drastic changes to the PR which is not present in the proposal.

Reply:
@abdulrahuman5196 The details api is needed and provides the missing address details i.e. long country name, streetNumber etc. The details api also provides longitude and latitude which can be used in generating the waypoint map for start and end address.

Reply:
> To fix the personal details search results breakage.

@abdulrahuman5196 The api call is not for resolving the search results breakage. The search results arrow key focus issue is resolved in the previous commit. The api call is needed so the personal details form is correctly populated i.e the country name and the streetNumber.